### DISCUSSION TITLE
How to override or edit Terms and Conditions?

### DISCUSSION THREAD

**Discussion by edemen** [keycloak/keycloak#22696] [Q&A]
Hi. I have recently upgraded from Keycloak 18 to 22 because I really needed the GDPR compliant terms checkbox at the initial registration screen.

I am installing with bitnami/keycloak helm, so my problem could be specific to this version only.
In KC 18 I used to be able to edit the file `/opt/bitnami/keycloak/themes/base/login/terms.ftl` with some super plain HTML to tailor the TaC text.

However, in KC 22 (and this is what I don't know whether it is purely because of bitnami or not), there are no theme files in the container, so I can't even edit anything. There is a README.md in the themes directory suggesting I create my own theme, but really this seems to be a massive over-kill just to have custom TaC text.

Is there a way to either pass it somehow during installation, with Env vars, as a single file in the themes directory or anything simple that does not require full on custom themes or building a custom image?

Thanks!

**Comment by justin-tay** [comment]
I think if you really need a one file solution then replacing the `lib/lib/main/org.keycloak.keycloak-themes-22.0.1.jar` is probably the way to go. It's not really recommended because of upgrades.

You don't really need that many files for the custom theme route. You do need to set the theme in the Realm settings though. `Themes > Login theme > mytheme`

```
themes
+ mytheme
  + login
    + theme.properties
    + messages
      + messages_en.properties
```

`theme.properties` (where `keycloak` is the parent theme you are extending)
```properties
parent=keycloak
import=common/keycloak
```

`messages_en.properties` (only the properties you wish to change needs to be included - you can localize the other languages as well with the respective postfixes eg `messages_fr.properties`. note that the default message resources are in the `base` theme)
```properties
termsTitle=My Terms and Conditions
termsText=<p>My Terms and conditions to be defined</p>
termsPlainText=My Terms and conditions to be defined.

**Reply by edemen** [reply]
This does not sounds too bad, I will try that, thanks!
